algorithms/2-search-solutions/fibonnaci_recursive.rb

```diff
@@ -0,0 +1,36 @@
+require 'benchmark'
+
+def fib(n)
+   if (n == 0)
+     # #1
+     return 0
+   elsif (n == 1)
+     # #2
+     return 1
+   else
+     # #3
+     return fib(n-1) + fib(n-2)
+   end
+ end
+
+ def iterative_fib(n)
+   if n == 0
+     return nil
+   elsif n == 1
+     return 0
+   else
+     counter = 1
+     ten_returned = [0, 1]
```

Comment:
Instead of an array of everything, you could have to variables for the last two that you update each iteration. It would probably be more efficient.